src/main/io/display.c

```diff
@@ -729,4 +744,56 @@ void displayDisablePageCycling(void)
     pageState.pageFlags &= ~PAGE_STATE_FLAG_CYCLE_ENABLED;
 }
 
+#ifdef OLEDCMS
+#include "io/cms.h"
+
+int displayCmsBegin(void)
+{
+    displayInCMS = true;
+
+    return 0;
+}
+
+int displayCmsEnd(void)
+{
+    displayInCMS = false;
+
+    return 0;
+}
+
+int displayCmsClear(void)
+{
+    i2c_OLED_clear_display_quick();
+
+    return 0;
+}
+
+int displayCmsWrite(uint8_t x, uint8_t y, char *s)
+{
+    i2c_OLED_set_xy(x, y);
+    i2c_OLED_send_string(s);
+
+    return 0;
+}
+
+screenFnVTable_t displayCmsVTable = {
+    displayCmsBegin,
+    displayCmsEnd,
+    displayCmsClear,
+    displayCmsWrite,
+    NULL,
+    NULL,
+    NULL,
+};
+
+void displayCmsInit(displayPort_t *pPort)
+{
+    pPort->rows = 8;
+    pPort->cols = 21;
+    pPort->buftime = 1;
+    pPort->bufsize = 50000;
```

Comment:
buftime and bufsize are both gone; they were used to estimate the tx buffer usage, but it's directly obtained by mspSerialTxBytesFree.